src/arrayMethodJoin.js

```diff
@@ -3,9 +3,45 @@
 /**
  * Implement method join
  */
+
+function separatorHandler(separator) {
+  let separate = separator;
+
+  if (separator === null) {
+    separate = 'null';
+  }
+
+  return separate;
+}
+
+function elementHandler(arrElement) {
+  let element = arrElement;
+
+  if (arrElement === null || arrElement === undefined) {
+    element = '';
+  }
+
+  return element;
+}
+
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const separate = separatorHandler(separator);
```

Comment:
Try to use native js and here you can use and check if this will work to your solution
```suggestion
    const separatorStr = String(separator);
```